sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyDoFnInvokerFactory.java

```diff
@@ -129,6 +130,32 @@ private ByteBuddyDoFnInvokerFactory() {}
     return newByteBuddyInvoker(DoFnSignatures.getSignature((Class) fn.getClass()), fn);
   }
 
+  /**
+   * Internal base class for generated {@link DoFnInvoker} instances.
+   *
+   * <p>This class should <i>not</i> be extended directly, or by Beam users. It must be public for
+   * generated instances to have adequate access, as they are generated "inside" the invoked {@link
+   * DoFn} class.
+   */
+  public abstract static class DoFnInvokerBase<InputT, OutputT, DoFnT extends DoFn<InputT, OutputT>>
+      implements DoFnInvoker<InputT, OutputT> {
+    protected DoFnT delegate;
+    private Map<String, OnTimerInvoker> onTimerInvokers = new HashMap<>();
+
+    public DoFnInvokerBase(DoFnT delegate) {
+      this.delegate = delegate;
+    }
+
+    void addOnTimerInvoker(String timerId, OnTimerInvoker onTimerInvoker) {
```

Comment:
Maybe add Javadoc about the reason you're pursuing this approach (possibly in the class or here). Essentially,

ByteBuddy doesn't like to generate conditional code, so you've decided to use a map + a lookup of the Timer ID to deal with that. Etc.